File: src/vinzi/test/TestClass.clj
(ns vinzi.test.TestClass
    (use vinzi.hib-connect.globals)
    (import vinzi.test.TestClass_jv))

(defrecord TestClass [id val1 val2])

(defn TestClass-to-java [cRec]
	(TestClass_jv. (:id cRec) (:val1 cRec) (:val2 cRec) ))


(add-to-java (TestClass. 0 1 2) TestClass-to-java)

(defn TestClass-to-clj [jRec]
	(TestClass. (.getId jRec) (.getVal1 jRec) (.getVal2 jRec) ))


(add-to-clj (TestClass_jv.) TestClass-to-clj)

File: src/java/TestClass.java
package vinzi.test;

public class TestClass {
    public int val1;
    public int getVal1() { return val1;}
    public void setVal1(int val1)  {this.val1 = val1; };


    public String val2;
    public String getVal2() { return val2;}
    public void setVal2(String val2)  {this.val2 = val2; };


    public TestClass() {};

    public TestClass(int val1, String val2) {
	this.val1 = val1;
	this.val2 = val2;
	return;};

   public String toString() {
        return "val1=" + this.val1 + "val2=" + this.val2;};

};
